### PR TITLE
Non-client library example of constructing a Signed URL

### PR DESCRIPTION
**Pull request by frankyn**
Example of how to construct a Google V4 Signed URL with an RSA signature using Python without a Cloud Client library.

### CODE REVIEW

File: storage/signed_urls/generate_signed_urls.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                        expiration, http_method='GET', query_parameters=None,
+                        headers=None):
+
+    if expiration > 604800:
+        print('Expiration Time can\'t be longer than 604800 seconds (7 days).')
+        sys.exit(1)
+
+    # [START storage_signed_url_canonical_uri]
+    escaped_object_name = quote(object_name, safe='')
+    canonical_uri = "/{}/{}".format(bucket_name, escaped_object_name)
```
Comment (gguuss) [inline storage/signed_urls/generate_signed_urls.py]: Prefer single-quote strings over double-quotes, there are a few more below to get too

**Comment by frankyn** [comment]
Thanks @gguuss! PTAL when you have a moment.

File: storage/signed_urls/generate_signed_urls.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+    # [START storage_signed_url_string_to_sign]
+    string_to_sign = '\n'.join(['GOOG4-RSA-SHA256',
+                                request_timestamp,
+                                credential_scope,
+                                canonical_request_hash])
+    # [END storage_signed_url_string_to_sign]
+
+    # [START storage_signed_url_signer]
+    signature = google_credentials.signer.sign(string_to_sign).hex()
```
Comment (gguuss) [inline storage/signed_urls/generate_signed_urls.py]: FYI your tests / code does not run on Python 2:

```
Traceback (most recent call last):
  File "/usr/local/google/home/class/code/samples/python-docs-samples/storage/signed_urls/generate_signed_urls_test.py", line 40, in test_generate_get_signed_url
    expiration=60)
  File "/usr/local/google/home/class/code/samples/python-docs-samples/storage/signed_urls/generate_signed_urls.py", line 133, in generate_signed_url
    signature = google_credentials.signer.sign(string_to_sign).hex()
AttributeError: 'str' object has no attribute 'hex'
```

**Comment by engelke** [comment]
Changed to be Python 2 and 3 compatible.

**Comment by frankyn** [comment]
Thanks @engelke for finishing up this request.

**Pull request merged**